okhttp/src/main/java/okhttp3/UnmodifiableList.java

```diff
@@ -0,0 +1,46 @@
+package okhttp3;
+
+import java.util.Iterator;
+import java.util.List;
+
+public final class UnmodifiableList<T> implements Iterable<T> {
```

Comment:
Yeah. Initially, I wanted to change `List` to `Iterable` but then found usages with `size()` and `get(index)` in the project, though they can be carefully rewritten.

Maybe you'll find such class useful for other cases when OkHttp uses/provides lists? (It will remove whole class of runtime problems).

Feel free to close this PR if you see a better solution (or it's not a problem).